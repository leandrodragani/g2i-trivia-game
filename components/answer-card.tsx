import React, { memo, useContext } from "react";
import { TouchableOpacity, TouchableOpacityProps } from "react-native";
import { ThemeContext } from "styled-components/native";
import { Ionicons } from "@expo/vector-icons";
import { Text } from "./text";
import { Box } from "./box";
import { AllHtmlEntities } from "html-entities";

export interface AnswerCardProps extends TouchableOpacityProps {
  answer: string;
  selected?: boolean;
  error?: boolean;
}

export const AnswerCard = memo(
  ({ answer, selected, error, ...props }: AnswerCardProps) => {
    const theme = useContext(ThemeContext);
    const entities = new AllHtmlEntities();
    const {
      colors: { red, green, gray, white },
      layout: { width },
      font: { medium },
    } = theme;

    return (
      <TouchableOpacity {...props} activeOpacity={0.75}>
        <Box
          flexDirection="row"
          border={1}
          width={width - 50}
          borderColor={gray[800]}
          bg={selected ? gray[800] : undefined}
          alignItems="center"
          justifyContent="space-between"
          borderRadius={8}
          padding={16}
        >
          <Text
            color={selected ? white : gray[500]}
            fontFamily={medium}
            fontSize={16}
          >
            {entities.decode(answer)}
          </Text>
          {selected ? (
            <Ionicons
              name="ios-checkmark-circle"
              size={20}
              color={green[500]}
            />
          ) : error ? (
            <Ionicons name="ios-close-circle" size={20} color={red[500]} />
          ) : null}
        </Box>
      </TouchableOpacity>
    );
  }
);
