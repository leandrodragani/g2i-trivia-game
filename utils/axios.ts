import axios from "axios";

export const apiInstance = axios.create({
  baseURL: "https://opentdb.com",
});

export const fetcher = (url: string) =>
  apiInstance.get(url).then((res) => res.data);
