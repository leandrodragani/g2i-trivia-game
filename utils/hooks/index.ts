export * from "./use-cached-resources";
export * from "./use-modal";
export * from "./use-before-leave";