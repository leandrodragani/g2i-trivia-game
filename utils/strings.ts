export function objectToQueryParams(object: any) {
  const sanitize = (object: any) => {
    Object.keys(object).forEach((key) =>
      object[key] === undefined || object[key] === null
        ? delete object[key]
        : {}
    );
    return object;
  };

  const queryString = new URLSearchParams(sanitize(object)).toString();
  return queryString;
}
