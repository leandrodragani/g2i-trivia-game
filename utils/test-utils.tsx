import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render, RenderOptions } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";
import theme from "styles/theme";
import { ToastProvider } from "context";
import { SafeAreaProvider } from "react-native-safe-area-context";

interface WrappedProvidersProps {
  children: React.ReactNode;
}

const WrappedProviders = ({ children }: WrappedProvidersProps) => {
  return (
    <ThemeProvider {...{ theme }}>
      <SafeAreaProvider>
        <NavigationContainer>
          <ToastProvider>{children}</ToastProvider>
        </NavigationContainer>
      </SafeAreaProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement<any>, options?: RenderOptions) =>
  render(ui, { wrapper: WrappedProviders, ...options });

export * from "@testing-library/react-native";

export { customRender as render };
