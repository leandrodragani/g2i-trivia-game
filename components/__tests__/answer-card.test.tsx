import React from "react";
import { render } from "utils/test-utils";
import { AnswerCard } from "../answer-card";

test("must show selected icon", async () => {
  const { toJSON, findByText, findByTestId } = render(
    <AnswerCard answer="Answer" selected />
  );

  const icon = findByTestId("selectedIcon");
  expect(icon).toBeTruthy();
  expect(findByText("Answer")).toBeTruthy();
  expect(toJSON).toMatchSnapshot();
});

test("must show error icon when answer is incorrect", async () => {
  const { findByTestId } = render(<AnswerCard answer="Answer" error />);

  const icon = findByTestId("errorIcon");
  expect(icon).toBeTruthy();
});
