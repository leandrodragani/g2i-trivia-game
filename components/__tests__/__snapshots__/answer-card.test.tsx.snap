// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get selected when i click it 1`] = `[Function]`;

exports[`must be selected when i press it 1`] = `[Function]`;
